pkg/pipeline/service.go

```diff
@@ -326,3 +329,26 @@ func (s *Service) notify(pipelineID string, err error) {
 		handler(e)
 	}
 }
+func (s *Service) validatePipeline(cfg Config, id string) error {
```

Comment:
I like how you extracted the validation logic into its own function. Can we do the same in the connector service please? 🙏